[#51] Display Profile photos in the inbox if sender has a Gmail/GSuite account

russelldc opened the pull request:
This is to solve Issue #51

### This isn't ready for merge yet, I just want input/ideas to solve a problem I'm encountering. @boukestam @abasiri @g60madman @ryanmclaughlin. Here's a copy/paste of my recent comment in that issue thread:

Alright, I've got the request all working, but I don't have it hooked up to anything automated/UI-related yet. It can be triggered from the Chrome web console, like:

```javascript
fetchProfilePhoto(['boukestam@gmail.com', 'noreply@medium.com', 'pay@amazon.com'])
```

You should see a parsed response like this one:
```javascript
[
   {
      "senderAddress": "joey@joeytroy.com",
      "photoURL": "https://lh3.googleusercontent.com/-XdUIqdMkC[....snipped...]"
   },
   {
      "senderAddress": "boukestam@gmail.com",
      "photoURL": "https://www.google.com/s2/photos/public/AIbEiAIAAABECJm0[....snipped...]"
   },
   {
      "senderAddress": "noreply@medium.com",
      "photoURL": "https://www.google.com/s2/photos/private/AIbEiAIAAABEiC3Zj[....snipped...]"
   },
   {
      "senderAddress": "pay@amazon.com",
      "photoURL":null
   }
]
```

I had to inject this code into the body/document, because of a CORS issue. Google's API wasn't accepting the requests from the content script (`script.js`), because they showed as originating from the extension URL.

I'm having trouble coming up with a sensible way to communicate between the injected script and the extension's content script.

Perhaps I'm overthinking it (very tired right now), and it doesn't need to communicate with the extension script at all?

abasiri commented on src/options/options.html:
```diff
@@ -18,21 +18,30 @@
         </div>
     </div>
+    <div class="setting">
+        <div class="setting-title">Profile Photos</div>
```
I would actually just have this be part of avatar's options. If avatar is enabled and picture is available, show picture, if picture not available show letter.

russelldc replied:
Oh? That setting was actually specifically inspired by you! Didn't you say something like: "I added this setting to turn avatars off because I just want profile photos anyway"?

russelldc replied:
Found it:

>I also added an option to disable showing the avatars (round circles with letters in them) I only ever found avatars useful when they had the photo of the person in them.

abasiri replied:
@russelldc yes, if profile pictures is working with avatars, I would turn avatars back on!

russelldc replied:
Ahh! :P I thought you didn't want the other avatars at all. Gotcha. I'll roll that back.

abasiri commented:
@russelldc Let me see if I understand the problem correctly first:

You have a list of emails in the extension script, but the code that gets the profile photos must be executed on the page. Now you need to send the list of emails from the extension script to the page script. And then send the URLs back. 

IF that is the issue you are trying to solve, there are a few potential approaches.

russelldc commented:
@abasiri Is the only thing keeping us from just injecting the entire extension script into the page the fact that we communicate with background.js for Options changes?

And yes, the thing I was hoping to do was to use this script as a satellite, just to fetch the photo URLs and get them back to the main script.

I *can* use localStorage between the two, I just can't use the 'storage' listener (because it only works between different tabs/pages), I'd have to manually poll for it.

abasiri commented:
No, background script is not stopping us from doing that (as far as I understand). But the background script is the main way to pass information from the content script to the page.

https://developers.chrome.com/extensions/messaging

abasiri commented:
I'm happy to play around with that as soon as I get a chance. That will probably be later this week.

Commit:
[#51] Fetch profile photos


Inject script directly, to avoid CORS issue. Add profile photo option


[#51] Adjust profile lookup request headers, remove unneeded code


[#51] Appends list of cachedProfiles with senderAddress, photoURL pairs


Consistently tabs rather than spaces

Commit:
Rebase on latest master, Combine profile photo option into avatar option

Commit:
Merge branch 'master' into profile-photo
